vignettes/GOstatsHyperG.Rmd

```diff
@@ -0,0 +1,521 @@
+|           genes. We recommend using the set of gene IDs that result from non-specific filtering. 
+|           The identifiers should be of the same type as the `geneIds`; for Affymetrix arrays, these will
+|           be Entrez Gene IDs.
+
+| **annotation** A string giving the name of the annotation data package that corresponds to the chip
+|           used in the experiment.
+
+| **ontology** A two-letter string specifying one of the three GO ontologies: BP, CC, or MF. 
+|          The `hyperGTest` function only tests a single GO ontology at one time.
+
+| **pvalueCutoff** A numeric values between zero and one used as a $p$-value cutoff for $p$-values
+|         generated by the Hypergeometric test. When the test being performed is non-conditional, this is
+|         only used as a default value for printing and summarizing the results. For a conditional
+|         analysis, the cutoff is used during the computation to determine perform the conditioning: child
+|         terms with a $p$-value less than `pvalueCutoff` are conditioned out of the test for their parent term.
+
+| **conditional** A logical value. If `TRUE`, the test performed uses the conditional algorithm. Otherwise,
+|         a standard Hypergeometric test is performed.When 'conditional(p) == TRUE', the 'hyperGTest'
+|         function uses the structure of the GO graph to estimate for each term whether or not there is
+|         evidence beyond that which is provided by the term's children to call the term in question
+|         statistically overrepresented. The algorithm conditions on all child terms that are themselves
+|         significant at the specified p-value cutoff. Given a subgraph of one of the three GO ontologies,
+|         the terms with no child categories are tested first. Next the nodes whose children have already
+|         been tested are tested. If any of a given node's children tested significant, the appropriate
+|         conditioning is performed.
+
+| **testDirection** A string which can be either "over" or "under". This determines whether the test
+|         performed detects over or under represented GO terms.
+:::
+
+```{r standardHyperGeo, echo=TRUE, results='hide', cache=TRUE}
+hgCutoff<- 0.001 
+params <- new("GOHyperGParams", 
+              geneIds=selectedEntrezIds,
+              universeGeneIds=entrezUniverse, 
+              annotation="hgu95av2.db",
+              ontology="BP", 
+              pvalueCutoff=hgCutoff, 
+              conditional=FALSE,
+              testDirection="over")
+```
+
+We would also like to perform a conditional test. Instead of having to
+define a new *GOHyperGParams* instance by hand, we can create a copy and
+update just the parameter of interest.
+
+```{r condHyperGeo, echo=TRUE, results='hide', cache=TRUE}
+paramsCond <- params 
+conditional(paramsCond) <- TRUE 
+```
+A similar approach would work to create a parameter object for testing a
```

Comment:
Insert a carriage return between the code block fence on 360 and `A similar approach` on 361.